lib/Frontend/nvqpp/ConvertExpr.cpp

```diff
@@ -2152,23 +2152,21 @@ bool QuakeBridgeVisitor::TraverseInitListExpr(clang::InitListExpr *x,
 
 bool QuakeBridgeVisitor::VisitInitListExpr(clang::InitListExpr *x) {
   auto loc = toLocation(x);
-  auto size = x->getNumInits();
+  std::int32_t size = x->getNumInits();
   auto initListTy = popType();
   if (size == 0) {
-    // TODO: Maybe check that this is an instance of a callable class, if it is
-    // a CXXRecordType.
+    // Nothing in the list. Just allocate the type.
     return pushValue(builder.create<cc::AllocaOp>(loc, initListTy));
   }
 
   // List has 1 or more members.
   auto last = lastValues(size);
-  bool allRef = [&]() {
-    for (auto v : last)
-      if (!isa<quake::RefType, quake::VeqType>(v.getType()))
-        return false;
-    return true;
-  }();
+  bool allRef = std::all_of(last.begin(), last.end(), [](auto v) {
+    return isa<quake::RefType, quake::VeqType>(v.getType());
+  });
   if (allRef) {
+    // Initializer list contains all quantum reference types. In this case we
+    // want to create quake code to concatenate the references into a veq.
```

Comment:
Is there a test for this type of quantum concatenation?

Reply:
I did a quick search and couldn't find it in the current main.